dex-it/src/test/scala/com/wavesplatform/it/sync/orders/MakerTakerFeeTestSuite.scala

```diff
@@ -0,0 +1,81 @@
+package com.wavesplatform.it.sync.orders
+
+import com.typesafe.config.{Config, ConfigFactory}
+import com.wavesplatform.dex.domain.asset.Asset.Waves
+import com.wavesplatform.dex.domain.order.OrderType.{BUY, SELL}
+import com.wavesplatform.it.MatcherSuiteBase
+
+class MakerTakerFeeTestSuite extends MatcherSuiteBase {
+
+  override protected val dexInitialSuiteConfig: Config = ConfigFactory.parseString(
+    s"""
+       |waves.dex {
+       |  price-assets = [ "$UsdId", "WAVES" ]
+       |  order-fee {
+       |    mode = dynamic
+       |    dynamic {
+       |      base-fee = 300000
+       |      zero-maker-double-taker = true
+       |    }
+       |  }
+       |}
+       """.stripMargin
+  )
+
+  override protected def beforeAll(): Unit = {
+    wavesNode1.start()
+    broadcastAndAwait(IssueUsdTx)
+    dex1.start()
+  }
+
+  private val doubledFee: Long = 2 * matcherFee
+  private val maker            = bob
+  private val taker            = alice
+
+  private def test(makerAmount: Long, takerAmount: Long, takerFee: Long, isTakerMarket: Boolean = false): Unit = {
+
+    dex1.api.cancelAll(maker)
+    dex1.api.cancelAll(taker)
+
+    val makerInitialBalance = wavesNode1.api.balance(maker, Waves)
+    val takerInitialBalance = wavesNode1.api.balance(taker, Waves)
+
+    val makerOrder = mkOrder(maker, wavesUsdPair, SELL, makerAmount, 300, doubledFee)
+    val takerOrder = mkOrder(taker, wavesUsdPair, BUY, takerAmount, 300, doubledFee)
+
+    placeAndAwaitAtDex(makerOrder)
+    placeAndAwaitAtNode(takerOrder, isMarketOrder = isTakerMarket)
+
+    val executedAmount = math.min(takerAmount, makerAmount)
+
+    withClue(s"maker amount = $makerAmount, taker${if (isTakerMarket) " (market)" else ""} amount = $takerAmount, taker fee = $takerFee\n") {
+      wavesNode1.api.balance(maker, Waves) shouldBe (makerInitialBalance - executedAmount)
+      wavesNode1.api.balance(taker, Waves) shouldBe (takerInitialBalance + executedAmount - takerFee)
+    }
+  }
+
+  "DEX should charge 0 fee for makers and doubled fee for takers" - {
+
```

Comment:
uncovered cases:
 - market order
 - partial fill
 - possibility of being the taker with zero waves balance
 - fee in non-waves asset 
 - taker become a maker (is it possible?)